amqp.go

```diff
@@ -113,6 +113,10 @@ func (e *Exchange) Publish(routingKey, message, requestID string) error {
 		Priority:     0,
 	}
 
+	if e.channel == nil {
+		return errors.New("Empty channel")
```

Comment:
Do this check first, before constructing `msg`. Also, maybe say `channel is nil`?